lib/module.js

```diff
@@ -13,7 +13,7 @@ module.exports = function (baseModule, moduleDefinition) {
     .keys()
     .each((attributeKey) => {
       const value = moduleDefinition[attributeKey];
-      moduleObject[attributeKey] = isGenerator(value) ? wrapInGenRunCallback(value.awaitify()) : value;
+      moduleObject[attributeKey] = (isGenerator(value) && !doNotWrapInGenRunCallback) ? wrapInGenRunCallback(value.awaitify()) : value;
```

Comment:
@mrlotfi you should consider the case when the value is a generator and doNotWrapInGenRunCallback is true. The correct implementation is like this:
```javascript
moduleObject[attributeKey] = isGenerator(value) ? doNotWrapInGenRunCallback ?  value.awaitify() : wrapInGenRunCallback(value.awaitify()) : value;
```